frontend/src/metabase-lib/v1/Question.ts

```diff
@@ -205,7 +216,13 @@ class Question {
   }
 
   setDatasetQuery(newDatasetQuery: DatasetQuery): Question {
-    return this.setCard(assoc(this.card(), "dataset_query", newDatasetQuery));
+    const card = this.card();
+    const newQuery = assocIn(
```

Comment:
@bshepherdson does `assocIn` check if the value is changed? If it always creates a new object, it could kill caching. Could you double check? And same for `new Question()`

Reply:
Ah, good point. I don't know the details of how `icepick` works. I'll make sure this code returns the original object if there's nothing to change.

Reply:
Ah, this is actually fine! `assocIn(x, ["some", "path"], "current value") === x`.

Reply:
Great! Thanks for checking